packages/core/src/utils/make-namespaced.ts

```diff
@@ -0,0 +1,152 @@
+ */
+export function makeNamespacedInput(input: SolcInput, output: SolcOutput): SolcInput {
+  const modifiedSources: Record<string, { content?: string }> = {};
+
+  for (const [sourcePath] of Object.entries(input.sources)) {
+    const source = input.sources[sourcePath];
+
+    if (source.content === undefined) {
+      modifiedSources[sourcePath] = source;
+      continue;
+    }
+
+    const orig = Buffer.from(source.content, 'utf8');
+
+    const modifications: Modification[] = [];
+
+    for (const node of output.sources[sourcePath].ast.nodes) {
+      if (isNodeType('ContractDefinition', node)) {
+        const contractDef = node;
+
+        // Remove any calls to parent constructors from the inheritance list
+        const inherits = contractDef.baseContracts;
+        for (const inherit of inherits) {
+          if (isNodeType('InheritanceSpecifier', inherit)) {
+            assert(inherit.baseName.name !== undefined);
+            modifications.push(makeReplace(inherit, orig, inherit.baseName.name));
+          }
+        }
+
+        const contractNodes = contractDef.nodes;
+        for (const contractNode of contractNodes) {
+          if (
+            isNodeType('FunctionDefinition', contractNode) ||
+            isNodeType('ModifierDefinition', contractNode) ||
+            isNodeType('VariableDeclaration', contractNode)
+          ) {
+            const doc = contractNode.documentation;
+            if (doc) {
+              modifications.push(makeDelete(doc, orig));
+            }
+            modifications.push(makeDelete(contractNode, orig));
+          } else if (isNodeType('UsingForDirective', contractNode)) {
+            modifications.push(makeDelete(contractNode, orig));
+          } else if (isNodeType('StructDefinition', contractNode)) {
+            const storageLocation = getStorageLocationAnnotation(contractNode);
+            if (storageLocation !== undefined) {
+              const structName = contractNode.name;
+              const variableName = `$${structName}_${(Math.random() * 1e6).toFixed(0)}`;
+              const insertText = ` ${structName} ${variableName};`;
+
+              modifications.push(makeInsertAfter(contractNode, insertText));
+            }
+          }
+        }
+      } else if (isNodeType('FunctionDefinition', node) || isNodeType('VariableDeclaration', node)) {
+        // Replace with a dummy variable of arbitrary type
+        const name = node.name;
+        const insertText = `uint256 constant ${name} = 0;`;
+        modifications.push(makeReplace(node, orig, insertText));
+      } else if (isNodeType('UsingForDirective', node)) {
+        modifications.push(makeDelete(node, orig));
+      }
+    }
+
+    modifiedSources[sourcePath] = { ...source, content: getModifiedSource(orig, modifications) };
+  }
+
+  return { ...input, sources: modifiedSources, settings: { ...input.settings, outputSelection: OUTPUT_SELECTION } };
```

Comment:
I'm thinking we might want to disable optimizations in the settings too. Since the output selection doesn't include any bytecode, I'm not sure if it will have any effect though. I'm checking with the Solidity team to confirm.

Reply:
Update: No need to disable optimizations, it should apparently not run unless we request codegen outputs.